File: 프로그래머스/0/181883. 수열과 구간 쿼리 1/수열과 구간 쿼리 1.java
import java.util.*;
class Solution {
    public int[] solution(int[] arr, int[][] queries) {
        int[] answer = {};
        
        List<Integer> list = new ArrayList<>();
        for(int a : arr) list.add(a);
        
        for(int i = 0; i < queries.length; i++){
            int s = queries[i][0];
            int e = queries[i][1];
            
            for(int j = s; j <= e; j++){
                list.set(j, list.get(j) + 1);
            }
        }
        
        answer = new int[list.size()];
        
        for(int k = 0; k < answer.length; k++){
            answer[k] = list.get(k);
        }
        
        return answer;
    }
}

File: 프로그래머스/0/181883. 수열과 구간 쿼리 1/수열과 구간 쿼리 1.c
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>

int* solution(int arr[], size_t arr_len, int** queries, size_t queries_rows, size_t queries_cols) {

    int* answer = (int*)malloc(arr_len * sizeof(int));
    int s, e;
    
    for(int i = 0; i < arr_len; i++){
        answer[i] = arr[i];
    }
    
    for(int j = 0; j < queries_rows; j++){
        s = queries[j][0];
        e = queries[j][1];
        for(; s <= e; s++){
            answer[s]++;
        }
    }
    
    return answer;
}